### PR TITLE
#167728011 Setting up coveralls

### PR DESCRIPTION
**Pull request by Wokoro**
**What does this PR do?**
  
Implementation to allow code test coverage report for the project using coveralls.

  **Description of Task to be completed?**
  * Added repository to coveralls online service so that repository can be tracked.
  * Added script to send the test reports generated locally or through a continuous integration platform e.g TravisCI to coveralls online service.
  * Added test coverage badge to `README.md`
  * Created coverall.yml file, to hold configuration data for coveralls service.

  **How should this be manually tested?**

  #### step 1
  clone repository
  `git clone git@github.com:andela/cyclops-backend.git`

  #### step 2
  Install npm packages.
  Navigate to root project directory and run `npm install`

  #### step 3
  Write unit tests for a given feature

  #### step 4
  Generate test reports

  #### step 5 
  Send reports generated to coveralls online service.
  Navigate to project root directory and run `npm run coveralls`

  #### step 6
  Refresh `README.md` and observe coverage result updated in badge.

  **Any background context you want to provide?**

  * npm packages most be install
  * test data must be generated.
  * Test report must be sent to coverall online service

  **What are the relevant stories?**
  [#167728011](https://www.pivotaltracker.com/story/show/167728011)

  **Screenshots (if appropriate)**

  none

  **Questions:**

  none

### CODE REVIEW

**Review by richienabuk — APPROVED** [review]

**Review by Kingobaino — APPROVED** [review]

**Review by justinefe — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -7,7 +7,9 @@
     "lint": "eslint",
     "test": "echo \"Error: no test specified\" && exit 1",
     "start": "nodemon --exec babel-node src/index.js",
-    "build": "babel src/index.js --out-dir build"
+    "build": "babel src/index.js --out-dir build",
+    "db:connect": "babel-node ./src/database/config",
```
Comment (chukwuemekachm) [inline package.json]: Why the `db:connent` script? @Wokoro.
What do we want to achieve with it?

**Review by chukwuemekachm — COMMENTED** [review]
Looks good @Wokoro.
I have a question below. 👇🏽

File: package.json (hunk 1 above)
Comment (Wokoro) [reply package.json]: It's redundant, I would remove it ASAP, it was passed down from initial project setup.

Thanks for pointing it out.

File: README.md [hunk 2]
```diff
@@ -1,8 +1,6 @@
 Barefoot Nomad - Making company travel and accomodation easy and convinient.
 =======
-[![Reviewed by Hound](https://img.shields.io/badge/Reviewed_by-Hound-8E64B0.svg)](https://houndci.com)
-[![Build Status](https://travis-ci.com/andela/cyclops-backend.svg?branch=develop)](https://travis-ci.com/andela/cyclops-backend)
-
+[![Reviewed by Hound](https://img.shields.io/badge/Reviewed_by-Hound-8E64B0.svg)](https://houndci.com) [![Coverage Status](https://coveralls.io/repos/github/andela/cyclops-backend/badge.svg?branch=develop)](https://coveralls.io/github/andela/cyclops-backend?branch=develop)
```
Comment (chukwuemekachm) [inline README.md:3]: Can you return the `Travis` badge here?

**Review by chukwuemekachm — CHANGES_REQUESTED** [review]
Hey @Wokoro.
Your PR just removed the Travis badge on our README. The `Coveralls` badge also doesn't display coverage.

File: README.md (hunk 2 above)
Comment (Wokoro) [reply README.md:3]: Sorry about the removal of the badge I would fix that.

As for reporting, I would have to edit travis.yml file to send coverage reports to coveralls from travis...

So I think the *setting unit test PR* should be merged into `develop` branch so there would be tests to report....

What do you think @chukwuemekachm

Comment (chukwuemekachm) [reply README.md:3]: I'll merge this now and @Abobos will resolve report the coverage results to coveralls.

**Review by chukwuemekachm — APPROVED** [review]
LGTM

**Pull request merged**